<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="None">
  
  
  <link rel="shortcut icon" href="img/favicon.ico">
  <title>Alliance Persistence Model</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="css/theme.css" />
  <link rel="stylesheet" href="css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Index";
    var mkdocs_page_input_path = "index.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="js/jquery-2.1.1.min.js" defer></script>
  <script src="js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href="." class="icon icon-home"> Alliance Persistence Model</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="./search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="home/">Home</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href=".">Index</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#classes">Classes</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#mixins">Mixins</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#slots">Slots</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#types">Types</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#built-in">Built in</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#defined">Defined</a>
    </li>
        </ul>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="credits/">Credits</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href=".">Alliance Persistence Model</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".">Docs</a> &raquo;</li>
    
      
    
    <li>Index</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="alliance-schema-prototype-schema">Alliance-Schema-Prototype schema</h1>
<p>Alliance Schema Prototype</p>
<h3 id="classes">Classes</h3>
<ul>
<li><a href="Allele/">Allele</a></li>
<li><a href="Assembly/">Assembly</a></li>
<li><a href="AuthorReference/">AuthorReference</a></li>
<li><a href="Chromosome/">Chromosome</a> - The ID of the landmark used to establish the coordinate system for the current feature.</li>
<li><a href="CrossReference/">CrossReference</a></li>
<li><a href="Gene/">Gene</a></li>
<li><a href="GenomicEntity/">GenomicEntity</a></li>
<li><a href="InformationContentEntity/">InformationContentEntity</a> - a piece of information that typically describes some topic of discourse or is used as support. Precedence of identifiers for references is as follows: PMID if available; DOI if not; actual alternate CURIE otherwise.<ul>
<li><a href="Reference/">Reference</a></li>
<li><a href="Resource/">Resource</a></li>
</ul>
</li>
<li><a href="Person/">Person</a></li>
<li><a href="Species/">Species</a></li>
<li><a href="Transcript/">Transcript</a></li>
<li><a href="Variant/">Variant</a> - Variant class</li>
<li><a href="VariantGeneConsequence/">VariantGeneConsequence</a></li>
<li><a href="VariantGenomicLocation/">VariantGenomicLocation</a></li>
<li><a href="VariantTranscriptConsequence/">VariantTranscriptConsequence</a> - Class for transcript-level VEP results</li>
</ul>
<h3 id="mixins">Mixins</h3>
<h3 id="slots">Slots</h3>
<ul>
<li><a href="PubMed_type/">PubMed type</a> - Type of information content entity as described by pub med.</li>
<li><a href="abstract/">abstract</a></li>
<li><a href="address/">address</a> - the particulars of the place where someone or an organization is situated.  For now, this slot is a simple text "blob" containing all relevant details of the given location for fitness of purpose. For the moment, this "address" can include other contact details such as email and phone number(?).</li>
<li><a href="allele_id/">allele id</a></li>
<li><a href="alliance_category/">alliance category</a></li>
<li><a href="amino_acid_reference/">amino acid reference</a> - reference genome amino acid sequence at variant position<ul>
<li><a href="variant_transcript_consequence_amino_acid_reference/">variant transcript consequence➞amino acid reference</a> - Amino acid sequence encoded by codon(s) in reference genome sequence altered by the variant</li>
</ul>
</li>
<li><a href="amino_acid_variant/">amino acid variant</a> - variant amino acid sequence at variant position<ul>
<li><a href="variant_transcript_consequence_amino_acid_variant/">variant transcript consequence➞amino acid variant</a> - Amino acid sequence encoded by codon(s) in variant sequence</li>
</ul>
</li>
<li><a href="assembly_id/">assembly id</a></li>
<li><a href="association_slot/">association slot</a> - any slot that relates an association to another entity<ul>
<li><a href="subject/">subject</a> - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.<ul>
<li><a href="variant_gene_consequence_subject/">variant gene consequence➞subject</a> - subject should be the variant identifier</li>
<li><a href="variant_genomic_location_subject/">variant genomic location➞subject</a> - subject should be the variant identifier</li>
<li><a href="variant_transcript_consequence_subject/">variant transcript consequence➞subject</a> - subject should be the variant identifier</li>
</ul>
</li>
</ul>
</li>
<li><a href="authors/">authors</a></li>
<li><a href="cdna_end/">cdna end</a> - end position of variation in cDNA coordinates</li>
<li><a href="cdna_start/">cdna start</a> - start position of variation in cDNA coordinates<ul>
<li><a href="variant_transcript_consequence_cdna_start/">variant transcript consequence➞cdna start</a> - Start position of variant in cDNA coordinates</li>
</ul>
</li>
<li><a href="cds_end/">cds end</a> - end position of variation in CDS coordinates</li>
<li><a href="cds_start/">cds start</a> - start position of variation in CDS coordinates<ul>
<li><a href="variant_transcript_consequence_cds_start/">variant transcript consequence➞cds start</a> - Start position of variant in CDS coordinates</li>
</ul>
</li>
<li><a href="chromosome_id/">chromosome id</a></li>
<li><a href="citation/">citation</a></li>
<li><a href="cnda_end/">cnda end</a> - End position of variant in cDNA coordinates<ul>
<li><a href="variant_transcript_consequence_cnda_end/">variant transcript consequence➞cnda end</a></li>
</ul>
</li>
<li><a href="cns_end/">cns end</a> - End position of variant in CDS coordinates<ul>
<li><a href="variant_transcript_consequence_cns_end/">variant transcript consequence➞cns end</a></li>
</ul>
</li>
<li><a href="codon_reference/">codon reference</a> - reference sequence of codon(s) affected by variation - bases outside of the variant region are in lower case, those within are in upper case (e.g. cTa)<ul>
<li><a href="variant_transcript_consequence_codon_reference/">variant transcript consequence➞codon reference</a> - Reference genome sequence of codon(s) altered by the variant.  Bases affected by the variant are given in upper case, bases flanking the variation are given in lower case</li>
</ul>
</li>
<li><a href="codon_variant/">codon variant</a> - variant sequence of codon(s) affected by variation - bases outside of the variant region are in lower case, those within are in upper case (e.g. cAa)<ul>
<li><a href="variant_transcript_consequence_codon_variant/">variant transcript consequence➞codon variant</a> - Sequence of codon(s) in variant sequence altered by the variant. Bases affected by the variant are given in upper case, bases flanking the variation are given in lower case</li>
</ul>
</li>
<li><a href="computed_gene/">computed gene</a></li>
<li><a href="copyright_date/">copyright date</a></li>
<li><a href="corresponding_author/">corresponding author</a></li>
<li><a href="creation_date/">creation date</a> - date on which an entity was created. This can be applied to nodes or edges</li>
<li><a href="cross_reference_id/">cross reference id</a></li>
<li><a href="cross_references/">cross references</a></li>
<li><a href="data_provider/">data provider</a> - MOD that provided data</li>
<li><a href="date_arrived_in_PubMed/">date arrived in PubMed</a> - Day in which a reference or resource was created in PUBMED. Only relevant for PUBMED references.</li>
<li><a href="date_last_modified/">date last modified</a> - Date on which an entity was last modified by the MOD.</li>
<li><a href="date_last_modified_in_PubMed/">date last modified in PubMed</a> - Date on which an entity was created.</li>
<li><a href="date_produced/">date produced</a> - Date on which an entity was produced.</li>
<li><a href="date_published/">date published</a> - Date on which an entity was created.</li>
<li><a href="date_produced/">date_produced</a> - date AGR data generated by MOD</li>
<li><a href="day_published/">day published</a> - Day in which an entity was created.</li>
<li><a href="description/">description</a> - a human-readable description of an entity</li>
<li><a href="display_name/">display name</a></li>
<li><a href="editors/">editors</a> - holds between a resource and a editor_resource</li>
<li><a href="end/">end</a> - The end of the feature in positive 1-based integer coordinates</li>
<li><a href="first_name/">first name</a></li>
<li><a href="from_resource/">from resource</a></li>
<li><a href="from_species/">from species</a> - holds between a genomic entity and a species</li>
<li><a href="gene_id/">gene id</a></li>
<li><a href="genomic_reference_sequence/">genomic reference sequence</a> - reference genome sequence at variation position</li>
<li><a href="genomic_variant_sequence/">genomic variant sequence</a> - variant genome sequence at variation position</li>
<li><a href="has_assembly/">has assembly</a></li>
<li><a href="hgvs_coding_nomenclature/">hgvs coding nomenclature</a> - HGVSc nomenclature for variation in coding sequence<ul>
<li><a href="variant_transcript_consequence_hgvs_coding_nomenclature/">variant transcript consequence➞hgvs coding nomenclature</a> - HGVS coding sequence (HGVSc) name</li>
</ul>
</li>
<li><a href="hgvs_nomenclature/">hgvs nomenclature</a> - HGVSg nomenclature for variant</li>
<li><a href="hgvs_protein_nomenclature/">hgvs protein nomenclature</a> - HGVSp nomenclature for variation in protein<ul>
<li><a href="variant_transcript_consequence_hgvs_protein_nomenclature/">variant transcript consequence➞hgvs protein nomenclature</a> - HGVS protein sequence (HGVSp) name</li>
</ul>
</li>
<li><a href="id/">id</a> - A unique identifier for a thing. Must be either a CURIE shorthand for a URI or a complete URI<ul>
<li><a href="reference_id/">reference➞id</a></li>
<li><a href="resource_id/">resource➞id</a> - Precedence of identifiers for references is as follows: PMID if available; DOI if not; actual alternate CURIE otherwise.</li>
</ul>
</li>
<li><a href="initials/">initials</a></li>
<li><a href="is_variant_of_allele/">is variant of allele</a></li>
<li><a href="is_variant_of_transcript/">is variant of transcript</a></li>
<li><a href="iso_abbreviation/">iso abbreviation</a></li>
<li><a href="issue_date/">issue date</a> - Date on which an entity was issued.</li>
<li><a href="issue_name/">issue name</a></li>
<li><a href="keywords/">keywords</a> - keywords tagging a publication</li>
<li><a href="last_modified/">last modified</a> - Date on which an entity was last modified in PubMed or in the MOD.</li>
<li><a href="last_name/">last name</a></li>
<li><a href="located_on/">located on</a></li>
<li><a href="medline_abbreviation/">medline abbreviation</a></li>
<li><a href="middle_names/">middle names</a></li>
<li><a href="mod_reference_types/">mod reference types</a> - List of types as assigned at a MOD.</li>
<li><a href="month_published/">month published</a> - Month in which an entity was created.</li>
<li><a href="name/">name</a> - a human-readable name for an entity</li>
<li><a href="negated/">negated</a> - if set to true, then the association is negated i.e. is not true</li>
<li><a href="note/">note</a></li>
<li><a href="object/">object</a> - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.<ul>
<li><a href="variant_gene_consequence_object/">variant gene consequence➞object</a> - object should be the gene identifier</li>
<li><a href="variant_genomic_location_object/">variant genomic location➞object</a> - object should be the chromosome identifier</li>
<li><a href="variant_transcript_consequence_object/">variant transcript consequence➞object</a> - object should be the transcript identifier</li>
</ul>
</li>
<li><a href="online_issn/">online issn</a></li>
<li><a href="orcid/">orcid</a></li>
<li><a href="padding_left/">padding left</a> - flanking sequence upstream of the variation position on the sense strand of the reference genome</li>
<li><a href="padding_right/">padding right</a> - flanking sequence downstream of the variation position on the sense strand of the reference genome</li>
<li><a href="page_areas/">page areas</a></li>
<li><a href="pages/">pages</a> - page number of source referenced for statement or publication</li>
<li><a href="person_id/">person id</a><ul>
<li><a href="person_person_id/">person➞person id</a></li>
</ul>
</li>
<li><a href="polyphen_prediction/">polyphen prediction</a> - PolyPhen-2 prediction</li>
<li><a href="polyphen_score/">polyphen score</a> - PolyPhen-2 score between 0 and 1</li>
<li><a href="predicate/">predicate</a> - A high-level grouping for the relationship type. This is analogous to category for nodes. In RDF, this corresponds to rdf:predicate and in Neo4j this corresponds to the relationship type.</li>
<li><a href="prefix/">prefix</a> - Denormalization to help with classifying types of crossReferences, distinguishing DOIs from PMC ids, etc.</li>
<li><a href="print_issn/">print issn</a></li>
<li><a href="protein_end/">protein end</a> - end position of variation in protein amino acid coordinates<ul>
<li><a href="variant_transcript_consequence_protein_end/">variant transcript consequence➞protein end</a> - End position of variant in amino acid sequence</li>
</ul>
</li>
<li><a href="protein_sequence/">protein sequence</a></li>
<li><a href="protein_start/">protein start</a> - start position of variation in protein amino acid coordinates<ul>
<li><a href="variant_transcript_consequence_protein_start/">variant transcript consequence➞protein start</a> - Start position of variant in amino acid sequence</li>
</ul>
</li>
<li><a href="publisher/">publisher</a></li>
<li><a href="qualifiers/">qualifiers</a> - This is the MeSH qualifier term that is optionally added to the descriptor term.</li>
<li><a href="reference_id/">reference id</a></li>
<li><a href="references/">references</a> - holds between an object and a reference</li>
<li><a href="release/">release</a> - MOD release ID</li>
<li><a href="resource_id/">resource id</a></li>
<li><a href="sift_prediction/">sift prediction</a> - SIFT prediction</li>
<li><a href="sift_score/">sift score</a> - SIFT score between 0 and 1</li>
<li><a href="start/">start</a> - The start of the feature in positive 1-based integer coordinates</li>
<li><a href="summary/">summary</a></li>
<li><a href="symbol/">symbol</a> - Symbol for a particular thing</li>
<li><a href="synonym/">synonym</a> - holds between a named thing and a synonym</li>
<li><a href="synonyms/">synonyms</a></li>
<li><a href="tags/">tags</a> - Optional ist of controlled vocabulary tags that give categories to each pub (ie: "can show images").</li>
<li><a href="title/">title</a> - A human readable title for a reference.<ul>
<li><a href="resource_title/">resource➞title</a> - the title of the publication</li>
</ul>
</li>
<li><a href="topics/">topics</a> - Connects an object to a vocabulary term or ontology that describes some aspect of the entity.</li>
<li><a href="transcript_id/">transcript id</a></li>
<li><a href="type/">type</a></li>
<li><a href="vep_consequence/">vep consequence</a></li>
<li><a href="vep_impact/">vep impact</a> - VEP predicted impact of variation on molecule</li>
<li><a href="volume/">volume</a></li>
<li><a href="volumes/">volumes</a></li>
<li><a href="year_published/">year published</a> - Year in which an entity was created.</li>
</ul>
<h3 id="types">Types</h3>
<h4 id="built-in">Built in</h4>
<ul>
<li><strong>Bool</strong></li>
<li><strong>Decimal</strong></li>
<li><strong>ElementIdentifier</strong></li>
<li><strong>NCName</strong></li>
<li><strong>NodeIdentifier</strong></li>
<li><strong>URI</strong></li>
<li><strong>URIorCURIE</strong></li>
<li><strong>XSDDate</strong></li>
<li><strong>XSDDateTime</strong></li>
<li><strong>XSDTime</strong></li>
<li><strong>float</strong></li>
<li><strong>int</strong></li>
<li><strong>str</strong></li>
</ul>
<h4 id="defined">Defined</h4>
<ul>
<li><a href="types/BiologicalSequence/">BiologicalSequence</a>  (<a href="types/String/">String</a>) </li>
<li><a href="types/Boolean/">Boolean</a>  (<strong>Bool</strong>)  - A binary (true or false) value</li>
<li><a href="types/Date/">Date</a>  (<strong>XSDDate</strong>)  - a date (year, month and day) in an idealized calendar</li>
<li><a href="types/Datetime/">Datetime</a>  (<strong>XSDDateTime</strong>)  - The combination of a date and time</li>
<li><a href="types/Decimal/">Decimal</a>  (<strong>Decimal</strong>)  - A real number with arbitrary precision that conforms to the xsd:decimal specification</li>
<li><a href="types/Double/">Double</a>  (<strong>float</strong>)  - A real number that conforms to the xsd:double specification</li>
<li><a href="types/Float/">Float</a>  (<strong>float</strong>)  - A real number that conforms to the xsd:float specification</li>
<li><a href="types/Integer/">Integer</a>  (<strong>int</strong>)  - An integer</li>
<li><a href="types/Ncname/">Ncname</a>  (<strong>NCName</strong>)  - Prefix part of CURIE</li>
<li><a href="types/Nodeidentifier/">Nodeidentifier</a>  (<strong>NodeIdentifier</strong>)  - A URI, CURIE or BNODE that represents a node in a model.</li>
<li><a href="types/Objectidentifier/">Objectidentifier</a>  (<strong>ElementIdentifier</strong>)  - A URI or CURIE that represents an object in the model.</li>
<li><a href="types/String/">String</a>  (<strong>str</strong>)  - A character string</li>
<li><a href="types/Time/">Time</a>  (<strong>XSDTime</strong>)  - A time object represents a (local) time of day, independent of any particular day</li>
<li><a href="types/Uri/">Uri</a>  (<strong>URI</strong>)  - a complete URI</li>
<li><a href="types/Uriorcurie/">Uriorcurie</a>  (<strong>URIorCURIE</strong>)  - a URI or a CURIE</li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="credits/" class="btn btn-neutral float-right" title="Credits">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="home/" class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="home/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="credits/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '.';</script>
    <script src="js/theme.js" defer></script>
      <script src="search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>

<!--
MkDocs version : 1.1.2
Build Date UTC : 2021-04-22 01:37:08.432943+00:00
-->
