<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Type: transcript level consequence - {{schema.description}}</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Type: transcript level consequence";
    var mkdocs_page_input_path = "TranscriptLevelConsequence.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> {{schema.description}}</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../home/">Home</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Index</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../credits/">Credits</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">{{schema.description}}</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Type: transcript level consequence</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="type-transcript-level-consequence">Type: transcript level consequence</h1>
<p>Class for transcript-level VEP results</p>
<p>URI: <a href="http://alliancegenome.org/TranscriptLevelConsequence">alliance:TranscriptLevelConsequence</a></p>
<p><img alt="img" src="http://yuml.me/diagram/nofunky;dir:TB/class/[Variant],[Transcript]&lt;affects%20transcript%201..*-++[TranscriptLevelConsequence&#124;amino_acid_reference:biological_sequence%20%3F;amino_acid_variant:biological_sequence%20%3F;codon_reference:string%20%3F;codon_variant:string%20%3F;cdna_start:integer%20%3F;cdna_end:integer%20%3F;cds_start:integer%20%3F;cds_end:integer%20%3F;protein_start:integer%20%3F;protein_end:integer%20%3F;hgvs_protein_nomenclature:string%20%3F;hgvs_coding_nomenclature:string%20%3F;cnda_end:string%20%3F;cns_end:string%20%3F;vep_consequence(i):string;vep_impact(i):string;polyphen_prediction(i):string%20%3F;polyphen_score(i):float%20%3F;sift_prediction(i):string%20%3F;sift_score(i):float%20%3F],[Variant]++-%20has%20transcript%20level%20consequence%200..*&gt;[TranscriptLevelConsequence],[MolecularConsequence]^-[TranscriptLevelConsequence],[Transcript],[MolecularConsequence]" /></p>
<h2 id="parents">Parents</h2>
<ul>
<li>is_a: <a href="../MolecularConsequence/">MolecularConsequence</a> - Class for molecular consequences calculated by the EnsEMBL Variant Effect Predictor (VEP)</li>
</ul>
<h2 id="referenced-by-class">Referenced by class</h2>
<ul>
<li><strong><a href="../Variant/">Variant</a></strong> <em><a href="../has_transcript_level_consequence/">has transcript level consequence</a></em>  <sub>0..<em></sub>  </em><em><a href="./">TranscriptLevelConsequence</a></em>*</li>
</ul>
<h2 id="attributes">Attributes</h2>
<h3 id="own">Own</h3>
<ul>
<li><a href="../cdna_end/">cdna end</a>  <sub>OPT</sub><ul>
<li>Description: end position of variation in cDNA coordinates</li>
<li>range: <a href="../types/Integer/">Integer</a></li>
</ul>
</li>
<li><a href="../cds_end/">cds end</a>  <sub>OPT</sub><ul>
<li>Description: end position of variation in CDS coordinates</li>
<li>range: <a href="../types/Integer/">Integer</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_affects_transcript/">transcript level consequence➞affects transcript</a>  <sub>1..*</sub><ul>
<li>Description: Transcript that is affected by the associated variant</li>
<li>range: <a href="../Transcript/">Transcript</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_amino_acid_reference/">transcript level consequence➞amino acid reference</a>  <sub>OPT</sub><ul>
<li>Description: Amino acid sequence encoded by codon(s) in reference genome sequence altered by the variant</li>
<li>range: <a href="../types/BiologicalSequence/">BiologicalSequence</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_amino_acid_variant/">transcript level consequence➞amino acid variant</a>  <sub>OPT</sub><ul>
<li>Description: Amino acid sequence encoded by codon(s) in variant sequence</li>
<li>range: <a href="../types/BiologicalSequence/">BiologicalSequence</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_cdna_start/">transcript level consequence➞cdna start</a>  <sub>OPT</sub><ul>
<li>Description: Start position of variant in cDNA coordinates</li>
<li>range: <a href="../types/Integer/">Integer</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_cds_start/">transcript level consequence➞cds start</a>  <sub>OPT</sub><ul>
<li>Description: Start position of variant in CDS coordinates</li>
<li>range: <a href="../types/Integer/">Integer</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_cnda_end/">transcript level consequence➞cnda end</a>  <sub>OPT</sub><ul>
<li>Description: End position of variant in cDNA coordinates</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_cns_end/">transcript level consequence➞cns end</a>  <sub>OPT</sub><ul>
<li>Description: End position of variant in CDS coordinates</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_codon_reference/">transcript level consequence➞codon reference</a>  <sub>OPT</sub><ul>
<li>Description: Reference genome sequence of codon(s) altered by the variant.  Bases affected by the variant are given in upper case, bases flanking the variation are given in lower case</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_codon_variant/">transcript level consequence➞codon variant</a>  <sub>OPT</sub><ul>
<li>Description: Sequence of codon(s) in variant sequence altered by the variant. Bases affected by the variant are given in upper case, bases flanking the variation are given in lower case</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_hgvs_coding_nomenclature/">transcript level consequence➞hgvs coding nomenclature</a>  <sub>OPT</sub><ul>
<li>Description: HGVS coding sequence (HGVSc) name</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_hgvs_protein_nomenclature/">transcript level consequence➞hgvs protein nomenclature</a>  <sub>OPT</sub><ul>
<li>Description: HGVS protein sequence (HGVSp) name</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_protein_end/">transcript level consequence➞protein end</a>  <sub>OPT</sub><ul>
<li>Description: End position of variant in amino acid sequence</li>
<li>range: <a href="../types/Integer/">Integer</a></li>
</ul>
</li>
<li><a href="../transcript_level_consequence_protein_start/">transcript level consequence➞protein start</a>  <sub>OPT</sub><ul>
<li>Description: Start position of variant in amino acid sequence</li>
<li>range: <a href="../types/Integer/">Integer</a></li>
</ul>
</li>
</ul>
<h3 id="inherited-from-molecular-consequence">Inherited from molecular consequence:</h3>
<ul>
<li><a href="../molecular_consequence_consequence_of_variant/">molecular consequence➞consequence of variant</a>  <sub>1..*</sub><ul>
<li>Description: Variant associated with the molecular consequence</li>
<li>range: <a href="../Variant/">Variant</a></li>
</ul>
</li>
<li><a href="../molecular_consequence_polyphen_prediction/">molecular consequence➞polyphen prediction</a>  <sub>OPT</sub><ul>
<li>Description: Pathogenicity prediction category calculated by PolyPhen-2</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../molecular_consequence_polyphen_score/">molecular consequence➞polyphen score</a>  <sub>OPT</sub><ul>
<li>Description: Pathogenicity score, between 0 and 1, calculated by PolyPhen-2</li>
<li>range: <a href="../types/Float/">Float</a></li>
</ul>
</li>
<li><a href="../molecular_consequence_sift_prediction/">molecular consequence➞sift prediction</a>  <sub>OPT</sub><ul>
<li>Description: Pathogenicity prediction category calculated by SIFT</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../molecular_consequence_sift_score/">molecular consequence➞sift score</a>  <sub>OPT</sub><ul>
<li>Description: Pathogenicity score calculated by SIFT</li>
<li>range: <a href="../types/Float/">Float</a></li>
</ul>
</li>
<li><a href="../molecular_consequence_vep_consequence/">molecular consequence➞vep consequence</a>  <sub>REQ</sub><ul>
<li>Description: EnsEMBL VEP calculated consequence (SO term)</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
<li><a href="../molecular_consequence_vep_impact/">molecular consequence➞vep impact</a>  <sub>REQ</sub><ul>
<li>Description: EnsEMBL VEP impact rating (HIGH/MODERATE/LOW/MODIFIER)</li>
<li>range: <a href="../types/String/">String</a></li>
</ul>
</li>
</ul>
              
            </div>
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
